server/routes/users.js

```diff
@@ -3,167 +3,473 @@ var express = require('express');
 var router = express.Router();
 
 // Require database adapter file (not node-postgres directly)
-const db = require('../db')
+var knex = require('../db/knex');
+var util = require('../util');
+var Promise = require('bluebird');
 
 // GET test string to verify that Users server is running.
 router.get('/ping', function(req, res, next) {
   res.send('pong - Users API');
 });
 
-// GET users listing.
+//GET all users.
 router.get('/', function(req, res, next) {
-  db.query('SELECT id FROM swe_user', [], (err, result) => {
-    if (err) return next(err);
-    res.send(result.rows);
-  });
+  knex('swe_user').select('id')
+    .then(result => {
+      if (result.length) {
+        res.json(result);
+      } else {
+        util.throwError(404, 'No users found');
+      }
+    })
+    .catch(err => { return next(err) });
 });
 
 // GET all user names.
 router.get('/names', function(req, res, next) {
-  db.query('SELECT CONCAT(first_name, \' \', last_name) FROM swe_user', [], (err, result) => {
-    if (err) return next(err);
-    res.send(result.rows);
-  });
+  knex('swe_user').select('first_name', 'last_name')
+    .then(result => {
+      if (result.length) {
+        res.json(result);
+      } else {
+        util.throwError(404, 'No user names found');
+      }
+    })
+    .catch(err => { return next(err) });
 });
 
 // GET all user emails.
 router.get('/emails', function(req, res, next) {
-  db.query('SELECT email FROM swe_user', [], (err, result) => {
-    if (err) return next(err);
-    res.send(result.rows);
-  });
+  knex('swe_user').select('email')
+    .then(result => {
+      if (result.length) {
+        res.json(result);
+      } else {
+        util.throwError(404, 'No user emails found');
+      }
+    })
+    .catch(err => { return next(err) });
 });
  
 // GET all user university IDs.
 router.get('/ids', function(req, res, next) {
-  db.query('SELECT university_id FROM swe_user;', [], (err, result) => {
-    if (err) return next(err);
-    res.send(result.rows);
-  });
+  knex('swe_user').select('university_id')
+    .then(result => {
+      if (result.length) {
+        res.json(result);
+      } else {
+        util.throwError(404, 'No user university IDs found');
+      }
+    })
+    .catch(err => { return next(err) });
+});
+
+// Add a user
+router.post('/register', function(req, res, next) {
+  const diet_id = req.query.diet_id;
+  const occupation_id = req.query.occupation_id;
+  const position_id = req.query.position_id;
+  const company_id = req.query.company_id;
+  const major_id = req.query.major_id;
+  const rank = req.query.rank;
```

Comment:
If these params are only used once or twice, it may be more convenient to just use "req.query.diet_id" in the code rather than store it in a var that's barely used

Reply:
fixed

Reply:
ok